### ISSUE TITLE
Decide which fluid js commands should accept a track name. 

### ISSUE THREAD

**Issue by CharlesHolbrow** [CharlesHolbrow/cybr#12]
some methods accept a trackname, and return a bundle

[track-drums909.js#L38](https://github.com/CharlesHolbrow/cybr/blob/78b5a07663814cff1b3fc416de40ef50e6a382a3/fluid-client/recipes/track-drums909.js#L38)
[midiclip.js#L58](https://github.com/CharlesHolbrow/cybr/blob/78b5a07663814cff1b3fc416de40ef50e6a382a3/fluid-client/src/fluid/midiclip.js#L58)
[fluidOsc.js#L8](https://github.com/CharlesHolbrow/cybr/blob/78b5a07663814cff1b3fc416de40ef50e6a382a3/fluid-client/src/fluidOsc.js#L8)

```node
drumTrack909(trackName); // from a recipe
fluid.midiclip.create(trackName, clipName, startBeats, lengthBeats, notes);
fluid.sample.insertWav (trackName, clipName, startBeats, fileName);
```
In the examples above, bundles include a track selection message, plus anything else needed to setup the insert. 

Other functions like [fluid.midiclip.note](https://github.com/CharlesHolbrow/cybr/blob/78b5a07663814cff1b3fc416de40ef50e6a382a3/fluid-client/src/fluid/midiclip.js#L32) expect that you prepare for the note insert yourself by selecting an audio track and a midi clip manually. 
```node
fluid.midiclip.note(noteNum, startTimeInWholeNotes, durationInWholeNotes, velocity);
```

At a minimum, I should make sure this is documented. It's probably best to decide on a standard for *how* to decide which commands are which. 

Considerations include:
- It's a little annoying to to have to re-select a track, ensuring that there are no spelling mistakes, etc. 
- It can also be convenient to just tell the command which track to put something on
- Remember: after the server handles a bundle, it clears all selections. So you have to re-select active tracks. 
- If you are inserting one thing, you are probably inserting multiple things. (`fluid.sample.insertWav` should probably not accept a track name, and should probably not output a bundle)

**Comment by CharlesHolbrow** [comment]
functions:
```javascript
audiotrack.select() //USES trackName 
audiotrack.insertWav() //USES trackName 
midiclip.clear()
midiclip.select()
midiclip.note() //uses selected audiotrack
midiclip.create() //USES trackName 
plugin.select() //uses selected audiotrack
plugin.setParam() //uses selected audiotrack
plugin.save()
plugin.load() //uses selected audiotrack
plugin.loadTrkpreset() //uses selected audiotrack
plugin.addpath()
sampler.add() //uses selected audiotrack
sampler.clearAll() //uses selected audiotrack
transport.play()
transport.stop()
transport.loop()
transport.to()
global.save()
global.cd()
global.activate()
```

Only two functions (insertWav and midiclip.create) actually use trackName. I can't really think of any compelling arguments for why we would opt to stick with keeping trackName as a parameter.

One problem I ran into when I was writing an example and reading examples is that I would often be confused whether or not I would have to select an audiotrack before I’d try to do anything, so maybe removing the feature to specify trackName would be a good idea for streamlining the workflow.

Maybe we can make a new function called “audiotrack.insertWavWithTrack” that also creates a track while inserting (same with midiclip.create) just in case the user is doing a lot of this. Otherwise, it would be more code efficient in general to select an audiotrack beforehand.

**Comment by zhiweigan** [comment]
Currently, the selectedAudioTrack pointer gets reset every time it receives a new bundle. I plan to change this so the nested arrays in the bundle no longer need to be spread. I will make another pull request once I have that functionality added.

**Comment by CharlesHolbrow** [comment]
Great. this closes #12
